codestodians/food-basket-front-end
# [Feature: #154513466] Update graphql error logging logic

#### What does this PR do?
- Deprecates the hack in logging `graphql` errors and updates the logic with proper confirguration.

#### Description of Task to be completed?
- Define `DEVELOPMENT` global variable in webpack config.
- Allow recognition of defined global variable by eslint.
- Update apollo client with global dev variable.

#### How should this be manually tested?
- Checkout into the `ft-fix-error-logging-hack-154513466` branch.
- Purposely introduce an error by supplying invalid credentials for login
- Observe that the errors still log to the browser console as before.

#### Any background context you want to provide?
- Webpack allows for the availability of global variables upon compile. This is achieved by the use of the `DefinePlugin`. Configuring variables to be available throughout the app thus becomes sensible here. 

#### What are the relevant pivotal tracker stories?
- [#154513466](https://www.pivotaltracker.com/story/show/154513466)
